### PR TITLE
Cleanup HDL generators

### PR DESCRIPTION
**Pull request by BFH-ktt1**

### CODE REVIEW

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/Hdl.java [hunk 1]
```diff
@@ -289,4 +292,111 @@ public static boolean writeArchitecture(String targetDirectory, ArrayList<String
...
+          netMap.put(sourceName, getBusNameContinues(comp, endIndex, theNets));
+        } else {
+          /* The last case, we have to enumerate through each bit */
+          if (isVhdl()) {
+            final var sourceNetName = new StringBuilder();
+            for (var bit = 0; bit < nrOfBits; bit++) {
+              /* First we build the Line information */
+              sourceNetName.setLength(0);
+              sourceNetName.append(String.format("%s(%d) ", sourceName, bit));
+              ConnectionPoint solderPoint = connectionInformation.get((byte) bit);
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/fpga/hdlgenerator/Hdl.java:338]: `final var`?

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/Hdl.java [hunk 2]
```diff
@@ -289,4 +292,111 @@ public static boolean writeArchitecture(String targetDirectory, ArrayList<String
     return FileWriter.writeContents(outFile, contents);
   }
 
+  public static Map<String, String> getNetMap(String sourceName, boolean floatingPinTiedToGround,
+      netlistComponent comp, int endIndex, Netlist theNets) {
+    var netMap = new HashMap<String, String>();
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/fpga/hdlgenerator/Hdl.java:297]: `final`?

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/ToplevelHdlGeneratorFactory.java [hunk 3]
```diff
@@ -232,4 +234,61 @@ public boolean hasLedArrayType(String type) {
     }
     return contents.get();
   }
+
+  private static ArrayList<String> getToplevelCode(MapComponent component) {
+    final var temp = new StringBuffer();
+    final var contents = new ArrayList<String>();
+    if (component.getNrOfPins() <= 0) {
+      Reporter.report.addError("BUG: Found a component with no pins");
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/fpga/hdlgenerator/ToplevelHdlGeneratorFactory.java:242]: If this is a `bug` and not regular message, the chance it occurs are low. Wouldn't it be better then to also add something like `Please report this occurence.` call to the message?

File: src/main/java/com/cburch/logisim/std/memory/ShiftRegisterHDLGeneratorFactory.java [hunk 4]
```diff
@@ -88,11 +88,11 @@ public void getGenerationTimeWiresPorts(Netlist theNetlist, AttributeSet attrs)
         if (nrOfBits == 1) {
           if (Hdl.isVhdl()) {
             for (var stage = 0; stage < nrOfStages; stage++)
-              map.putAll(GetNetMap(String.format("D(%d)", stage), true, comp, 6 + 2 * stage, nets));
+              map.putAll(Hdl.getNetMap(String.format("D(%d)", stage), true, comp, 6 + 2 * stage, nets));
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/std/memory/ShiftRegisterHDLGeneratorFactory.java:91]: I'd put multiplication in `()` (same in line below)

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/Hdl.java (hunk 1 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/fpga/hdlgenerator/Hdl.java:338]: I missed that one, thanks! done

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/ToplevelHdlGeneratorFactory.java (hunk 3 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/fpga/hdlgenerator/ToplevelHdlGeneratorFactory.java:242]: Done

File: src/main/java/com/cburch/logisim/std/memory/ShiftRegisterHDLGeneratorFactory.java (hunk 4 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/std/memory/ShiftRegisterHDLGeneratorFactory.java:91]: done

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/HdlPorts.java [hunk 5]
```diff
@@ -170,6 +170,12 @@ public boolean doPullDownOnFloat(String name) {
     throw new ArrayStoreException("port not contained in structure");
   }
 
+  public boolean contains(String name) {
+    for (var port : myPorts)
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/fpga/hdlgenerator/HdlPorts.java:174]: `final var`

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java [hunk 6]
```diff
@@ -639,176 +641,127 @@ public String getRelativeDirectory() {
 
   private ArrayList<String> getVHDLBlackBox(Netlist theNetlist, AttributeSet attrs,
       String componentName, Boolean isEntity) {
-    var contents = new ArrayList<String>();
-    var oneLine = new StringBuilder();
-    var identSize = 0;
-    var compTab = (isEntity) ? "" : "   ";
-    var first = true;
+    var contents = LineBuffer.getHdlBuffer();
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java:644]: `final`?

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java [hunk 7]
```diff
@@ -639,176 +641,127 @@ public String getRelativeDirectory() {
...
-      contents.add("   COMPONENT " + componentName);
-    }
+    contents.add("{{1}} {{2}}{{3}}", isEntity ? "ENTITY" : "COMPONENT", componentName, isEntity ? " IS" : "");
     if (!myParametersList.isEmpty(attrs)) {
-      oneLine.append(compTab).append("   GENERIC ( ");
-      identSize = oneLine.length();
-      first = true;
+      // first we build a list with parameters to determine the max. string length
+      final var myParameters = new HashMap<String, Boolean>();
       for (var generic : myParametersList.keySet(attrs)) {
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java:656]: `final var generic`

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/HdlPorts.java (hunk 5 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/fpga/hdlgenerator/HdlPorts.java:174]: done

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java (hunk 6 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java:641]: done

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java (hunk 7 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java:653]: done

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java [hunk 8]
```diff
@@ -513,7 +510,7 @@ else if (componentInfo instanceof netlistComponent) {
       contents.add(oneLine.toString());
     }
     contents.add("");
-    return contents.getWithIndent(3);
+    return contents.getWithIndent(1);
```
Comment (MarcinOrlowski) [inline src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java:513]: `1` can be removed -> `getWithIndent()`

**Review by MarcinOrlowski — CHANGES_REQUESTED** [review]

File: src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java (hunk 8 above)
Comment (BFH-ktt1) [reply src/main/java/com/cburch/logisim/fpga/hdlgenerator/AbstractHdlGeneratorFactory.java]: done

**Review by MarcinOrlowski — APPROVED** [review]

**Pull request merged**